### PR TITLE
[feat-5485]: street name only search filterform 

### PR DESCRIPTION
**Pull request by vdegraaf**
Context
Feedback from testers was that beaviour of showing address in inputfield inconsistent was. Sometimes it showed just the streetname, sometimes it showed the streetname and the municipality. Also discovered that clicking Nieuw Filter didn't clear the value of the inputfield.
Now only unique streetnames are shown

Changes
dependent on prop streetNameOnly will serviceParams and formatPdokResponse change their behaviour.
Filtering for unique streetnames
removed defaultvalue from Autosuggest. -> Autosuggest seems to be a mix of uncontrolled and controlled feautures (with a useState to set the defaultvalue and an inPutRef). Story has been created (https://gemeente-amsterdam.atlassian.net/browse/SIG-5574) in Jira for refactoring.
tests for meeting threshold added

For testers
PDOK sends back all matches, so also the matches with the municipality. This shows most when searching for Ams. Can't do nothing about that.

Ticket: [SIG-5485](https://gemeente-amsterdam.atlassian.net/browse/SIG-5485)

[SIG-5485]: https://gemeente-amsterdam.atlassian.net/browse/SIG-5485?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/PDOKAutoSuggest/PDOKAutoSuggest.test.tsx [hunk 1]
```diff
@@ -43,6 +43,28 @@ const renderAndSearch = async (value = 'Dam', props = {}) => {
   await screen.findByTestId('auto-suggest')
 }
 
+const renderAndSearchWithStreetNameOnlyAsTrue = async (
+  value = 'dor',
+  props = {}
+) => {
+  render(
+    withAppContext(
+      <PDOKAutoSuggest onSelect={onSelect} streetNameOnly={true} {...props} />
```
Comment (OscarBakker) [inline src/components/PDOKAutoSuggest/PDOKAutoSuggest.test.tsx:52]: you can leave {true}

Comment (OscarBakker) [reply src/components/PDOKAutoSuggest/PDOKAutoSuggest.test.tsx:52]: why not add a third arg to renderAndSearch with streetNameOnly?

**Review by OscarBakker — CHANGES_REQUESTED** [review]

File: src/components/PDOKAutoSuggest/PDOKAutoSuggest.test.tsx (hunk 1 above)
Comment (SireeKoolenWijkstra) [reply src/components/PDOKAutoSuggest/PDOKAutoSuggest.test.tsx:52]: Euh, isn't streetNameOnly by default false and that's why I set it here expressly to true?

**Review by vdegraaf — COMMENTED** [review]
@SireeKoolenWijkstra wasn't it a criteria that only unique street names were shown in the dropdown?

**Review by vdegraaf — COMMENTED** [review]
Looks good! But cannot approve because I made the PR. -> Nice! Ik zal hem mergen

**Review by SireeKoolenWijkstra — APPROVED** [review]
lgt Vincent, dus heb ik de PR gemerged.

**Pull request merged**